### PR TITLE
Breadcrumbs on taxon pages

### PR DESCRIPTION
**Pull request by rubenarakelyan**
* Re-architect the breadcrumb functionality into a separate class to make it testable in isolation, and also show breadcrumbs on taxon pages.
* Add Jasmine tests for breadcrumb functionality.

### CODE REVIEW

File: app/routes.js [hunk 1]
```diff
@@ -6,30 +6,36 @@ var expressNunjucks = require('express-nunjucks');
...
+var BreadcrumbMaker = require('../lib/js/breadcrumb_maker.js');
 
 (function(){
   "use strict";
 
-  var metadata = getMetadata();
+  var metadata;
+  var breadcrumbMaker;
+  getMetadata().then(function(metadataResult) {
+    console.log('loading', metadataResult === undefined);
```
Comment (MatMoore) [inline app/routes.js]: Probably not! Let's remove it

File: app/routes.js [hunk 2]
```diff
@@ -6,30 +6,36 @@ var expressNunjucks = require('express-nunjucks');
 var Promise = require('bluebird');
 var glob = Promise.promisify(require('glob'));
 var readFile = Promise.promisify(fs.readFile);
+var BreadcrumbMaker = require('../lib/js/breadcrumb_maker.js');
```
Comment (MatMoore) [inline app/routes.js]: The `js` is there because we started using `lib/ruby` to organise the ruby code

Comment (carolinegreen) [inline app/routes.js]: Ah okay, fair.

File: app/routes.js (hunk 1 above)
Comment (rubenarakelyan) [inline app/routes.js]: Removed in the latest version

File: test/app/breadcrumb_maker_test.js [hunk 3]
```diff
@@ -0,0 +1,57 @@
+var assert = require('assert');
```
Comment (rubenarakelyan) [inline test/app/breadcrumb_maker_test.js]: OK, based on this comment I have changed our testing framework to Jasmine. Thankfully it was a pretty small change and our existing tests continue to work with minimal changes.

Comment (carolinegreen) [inline test/app/breadcrumb_maker_test.js]: 👍

**Pull request merged**